applications/RANSApplication/python_scripts/formulations/fractional_step/fractional_step_velocity_pressure_rans_formulation.py

```diff
@@ -0,0 +1,344 @@
+        CalculateNormalsOnConditions(model_part)
+
+        process_info = model_part.ProcessInfo
+        wall_model_part_name = process_info[KratosRANS.WALL_MODEL_PART_NAME]
+        kappa = process_info[KratosRANS.WALL_VON_KARMAN]
+        beta = process_info[KratosRANS.WALL_SMOOTHNESS_BETA]
+        wall_function_update_process = KratosRANS.RansWallFunctionUpdateProcess(
+            model_part.GetModel(),
+            wall_model_part_name,
+            kappa,
+            beta,
+            self.echo_level)
+
+        self.AddProcess(wall_function_update_process)
+
+        if (self.IsPeriodic()):
+            InitializePeriodicConditions(
+                model_part,
+                self.fractional_step_model_part,
+                None,
+                "FSPeriodicCondition{0:d}D".format(self.GetDomainSize()))
+
+        settings = self.GetParameters()
+
+        self.solver_settings = self.__CreateSolverSettings(
+            self.fractional_step_model_part,
+            self.GetDomainSize(),
+            settings["time_order"].GetInt(),
+            True,
+            self.GetMoveMeshFlag(),
+            settings["reform_dofs_at_each_step"].GetBool())
+
+        self.solver_settings.SetEchoLevel(self.echo_level)
+
+        ## Construct the linear solvers
+        linear_solver_factory = GetKratosObjectType("LinearSolverFactory")
+        pressure_linear_solver = linear_solver_factory(settings["pressure_linear_solver_settings"])
+        velocity_linear_solver = linear_solver_factory(settings["velocity_linear_solver_settings"])
+
+        strategy_label_type = GetKratosObjectType("StrategyLabel")
+
+        self.solver_settings.SetStrategy(
+            strategy_label_type.Velocity,
+            velocity_linear_solver,
+            settings["velocity_tolerance"].GetDouble(),
+            settings["maximum_velocity_iterations"].GetInt())
+        self.solver_settings.SetStrategy(
+            strategy_label_type.Pressure,
+            pressure_linear_solver,
+            settings["pressure_tolerance"].GetDouble(),
+            settings["maximum_pressure_iterations"].GetInt())
+
+        if self.IsPeriodic():
+            self.solver = GetKratosObjectType("FractionalStepStrategy")(
+                self.fractional_step_model_part,
+                self.solver_settings,
+                settings["predictor_corrector"].GetBool(),
+                False,
+                KratosCFD.PATCH_INDEX)
+        else:
+            self.solver = GetKratosObjectType("FractionalStepStrategy")(
+                self.fractional_step_model_part,
+                self.solver_settings,
+                settings["predictor_corrector"].GetBool(),
+                False)
+
+        process_info.SetValue(Kratos.DYNAMIC_TAU, settings["dynamic_tau"].GetDouble())
+        process_info.SetValue(Kratos.OSS_SWITCH, settings["oss_switch"].GetInt())
+
+        super().Initialize()
+
+        Kratos.Logger.PrintInfo(self.GetName(), "Solver initialization finished.")
```

Comment:
```suggestion
        Kratos.Logger.PrintInfo(self.__class__.__name__, "Solver initialization finished.")
```
With this you can remove the `GetName()` method.

Reply:
I am using the `GetName` method to provide the final class name, even when logger is writing from the base class.

Reply:
Yes. But I think this is exactly the idea. Then you know in which file your run is eventually crashing...

Reply:
I get your point, and it is done :)